pkg/network/common/common.go

```diff
@@ -224,3 +225,19 @@ func GetHostIPNetworks(skipInterfaces []string) ([]*net.IPNet, []net.IP, error)
 	}
 	return hostIPNets, hostIPs, kerrors.NewAggregate(errList)
 }
+
+func HSEgressIPsToStrings(ips []networkv1.HostSubnetEgressIP) []string {
```

Comment:
Sigh. This is a mess and we should revert the `openshift/api` change that required it.
(At a minimum, split the egressip changes into a separate commit so they're easily revertable when we fix this later.)

Reply:
In case someone else gets here later and also wonders the rationale for this:
https://github.com/openshift/api/commit/a63a88e3fd742e3522b20f47b40e38336570dc25

Reply:
grump: go has exactly two "magic" types. Why it can't implement type conversion across literally the only two polymorphic types in the whole world is infuriating.

Reply:
@abhat can fill us in, but I believe the issue is that it's not possible to add validation to an array

Reply:
Yes, but we don't *need* validation there if it's just going to make our lives more difficult. We have to validate the network CRD types at runtime anyway because the CRD validation only handles part of it. There is talk about adding a ValidatingWebHook at some point so we can do this properly.